diff --git a/lib/typescript/commonjs/src/navigators/createNativeStackNavigator.d.ts b/lib/typescript/commonjs/src/navigators/createNativeStackNavigator.d.ts
index 6b684813ba3d79a72fc29c8a0fbcc5cb31f5329e..0b7b13b1804817d66374e5e1f56e67294f50b5ad 100644
--- a/lib/typescript/commonjs/src/navigators/createNativeStackNavigator.d.ts
+++ b/lib/typescript/commonjs/src/navigators/createNativeStackNavigator.d.ts
@@ -1,7 +1,7 @@
 import { type NavigatorTypeBagBase, type ParamListBase, type StackNavigationState, type StaticConfig, type TypedNavigator } from '@react-navigation/native';
 import type { NativeStackNavigationEventMap, NativeStackNavigationOptions, NativeStackNavigationProp, NativeStackNavigatorProps } from '../types';
 declare function NativeStackNavigator({ id, initialRouteName, children, layout, screenListeners, screenOptions, screenLayout, UNSTABLE_getStateForRouteNamesChange, ...rest }: NativeStackNavigatorProps): import("react/jsx-runtime").JSX.Element;
-export declare function createNativeStackNavigator<const ParamList extends ParamListBase, const NavigatorID extends string | undefined = undefined, const TypeBag extends NavigatorTypeBagBase = {
+export declare function createNativeStackNavigator<ParamList extends ParamListBase, NavigatorID extends string | undefined = undefined, TypeBag extends NavigatorTypeBagBase = {
     ParamList: ParamList;
     NavigatorID: NavigatorID;
     State: StackNavigationState<ParamList>;
@@ -11,6 +11,6 @@ export declare function createNativeStackNavigator<const ParamList extends Param
         [RouteName in keyof ParamList]: NativeStackNavigationProp<ParamList, RouteName, NavigatorID>;
     };
     Navigator: typeof NativeStackNavigator;
-}, const Config extends StaticConfig<TypeBag> = StaticConfig<TypeBag>>(config?: Config): TypedNavigator<TypeBag, Config>;
+}, Config extends StaticConfig<TypeBag> = StaticConfig<TypeBag>>(config?: Config): TypedNavigator<TypeBag, Config>;
 export {};
 //# sourceMappingURL=createNativeStackNavigator.d.ts.map
\ No newline at end of file
diff --git a/src/navigators/createNativeStackNavigator.tsx b/src/navigators/createNativeStackNavigator.tsx
index c0ef42b70d10f3f3488f7a95354dcb4cca259d73..541dfc40abf14e084349704cf1d49755271a665f 100644
--- a/src/navigators/createNativeStackNavigator.tsx
+++ b/src/navigators/createNativeStackNavigator.tsx
@@ -91,9 +91,9 @@ function NativeStackNavigator({
 }
 
 export function createNativeStackNavigator<
-  const ParamList extends ParamListBase,
-  const NavigatorID extends string | undefined = undefined,
-  const TypeBag extends NavigatorTypeBagBase = {
+  ParamList extends ParamListBase,
+  NavigatorID extends string | undefined = undefined,
+  TypeBag extends NavigatorTypeBagBase = {
     ParamList: ParamList;
     NavigatorID: NavigatorID;
     State: StackNavigationState<ParamList>;
@@ -108,7 +108,7 @@ export function createNativeStackNavigator<
     };
     Navigator: typeof NativeStackNavigator;
   },
-  const Config extends StaticConfig<TypeBag> = StaticConfig<TypeBag>,
+  Config extends StaticConfig<TypeBag> = StaticConfig<TypeBag>,
 >(config?: Config): TypedNavigator<TypeBag, Config> {
   return createNavigatorFactory(NativeStackNavigator)(config);
 }
