diff --git a/node_modules/react-native-progress/Bar.js b/node_modules/react-native-progress/Bar.js
index dd8e877..8444efc 100644
--- a/node_modules/react-native-progress/Bar.js
+++ b/node_modules/react-native-progress/Bar.js
@@ -25,6 +25,7 @@ export default class ProgressBar extends Component {
     useNativeDriver: PropTypes.bool,
     animationConfig: PropTypes.object,
     animationType: PropTypes.oneOf(['decay', 'timing', 'spring']),
+    innerProgress: PropTypes.any,
   };
 
   static defaultProps = {
@@ -127,6 +128,7 @@ export default class ProgressBar extends Component {
       style,
       unfilledColor,
       width,
+      innerProgress,
       ...restProps
     } = this.props;
 
@@ -140,8 +142,13 @@ export default class ProgressBar extends Component {
       backgroundColor: unfilledColor,
     };
     const progressStyle = {
+      ...innerProgress,
       backgroundColor: color,
       height,
+      width: this.state.progress.interpolate({
+        inputRange: [0, 1],
+        outputRange: [0.0001, innerWidth]
+      }),
       transform: [
         {
           translateX: this.state.animationValue.interpolate({
@@ -149,19 +156,6 @@ export default class ProgressBar extends Component {
             outputRange: [innerWidth * -INDETERMINATE_WIDTH_FACTOR, innerWidth],
           }),
         },
-        {
-          translateX: this.state.progress.interpolate({
-            inputRange: [0, 1],
-            outputRange: [innerWidth / (I18nManager.isRTL ? 2 : -2), 0],
-          }),
-        },
-        {
-          // Interpolation a temp workaround for https://github.com/facebook/react-native/issues/6278
-          scaleX: this.state.progress.interpolate({
-            inputRange: [0, 1],
-            outputRange: [0.0001, 1],
-          }),
-        },
       ],
     };
 
