diff --git a/AnySizeDragSortableView.js b/AnySizeDragSortableView.js
index 9450cbc6782441c62fcb950491eca921cce72aee..c5b29f0373e727fb58f6bc4ce19a74063993dea2 100644
--- a/AnySizeDragSortableView.js
+++ b/AnySizeDragSortableView.js
@@ -1,16 +1,15 @@
 import React from 'react';
 import {
-  NativeModules,
   StyleSheet,
   ScrollView,
   View,
   PanResponder,
   LayoutAnimation,
-  Platform
+  Platform,
+  UIManager
 } from 'react-native';
 const PropTypes = require('prop-types')
 const ANIM_DURATION = 300
-const { UIManager } = NativeModules;
 
 if (Platform.OS === 'android') {
   if (UIManager.setLayoutAnimationEnabledExperimental) {
