diff --git a/RNGestureHandler.podspec b/RNGestureHandler.podspec
index 1ae15b20db8bf8491547f4e172deb531dcd40477..9e61d56c0a80ca881dedc812d3c22ec4b07152cf 100644
--- a/RNGestureHandler.podspec
+++ b/RNGestureHandler.podspec
@@ -29,4 +29,12 @@ Pod::Spec.new do |s|
   else
     s.dependency "React-Core"
   end
+
+  if ENV['USE_FRAMEWORKS'] != nil && ENV['RCT_NEW_ARCH_ENABLED'] == '1'
+    add_dependency(s, "React-FabricComponents", :additional_framework_paths => [
+      "react/renderer/textlayoutmanager/platform/ios",
+      "react/renderer/components/textinput/platform/ios",
+    ])
+    add_dependency(s, "React-rendererconsistency")
+  end
 end